build.gradle

```diff
@@ -16,9 +17,15 @@ repositories {
 }
 
 dependencies {
-    compile 'org.codehaus.groovy:groovy-all:2.3.11'
+    compile 'org.codehaus.groovy:groovy-all:2.5.12'
     compile 'info.picocli:picocli-groovy:4.1.4'
+    compile 'org.apache.maven:maven-core:3.6.3'
+    compile 'org.apache.maven:maven-model:3.6.3'
+
     testCompile group: 'junit', name: 'junit', version: '4.12'
+    testCompile platform("org.spockframework:spock-bom:2.0-M1-groovy-2.5")
+    testCompile "org.spockframework:spock-core"
+    testCompile "org.spockframework:spock-junit4"
```

Comment:
We should try and have some consistency with how we define the dependencies.  String, GString, map are all used here.  

Also is there any reason not to go with groovy 3 which is the current stable release for jdk8+?  We should make sure we're on the latest of all dependencies since this is greenfield development, no reason to start out behind.

Reply:
I think there was some issues with spock on groovy 3.  I'll take a look though.

Reply:
OK Let me know if there is, there is definitely a new release for groovy 3 (https://mvnrepository.com/artifact/org.spockframework/spock-bom/2.0-M3-groovy-3.0)

Reply:
Passed on new groovy version 3.0.4 and spock version 2.0-M3-groovy-3.0 to Nisarg to include in his unit test code...